### PR TITLE
Submit bloom filter as kinto attachment with cron job

### LINKED ISSUE #13616: Generate MLBF for v3 blocklist and submit to kinto as attachment
We need to build on the proof of concept for #13408 to save the blocklist as an attachment to a kinto record.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #13616 - this is using the (previously assumed to be enough) salt in the keys and the existing hash function the library provides.  I'll make the change to whatever the final encoding/hashing solution is in a follow-up.  Also, @Rob--W has requested a reliable date be included in the record metadata to compare signed dates too, which will also be added in a follow-up.

### CODE REVIEW

File: src/olympia/blocklist/cron.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+log = olympia.core.logger.getLogger('z.cron')
+
+
+def upload_mlbf_to_kinto():
+    server = KintoServer(KINTO_BUCKET, KINTO_COLLECTION_MLBF)
+    stats = {}
+    key_format = get_mlbf_key_format()
+    bloomfilter = generate_mlbf(stats, key_format)
+    with tempfile.NamedTemporaryFile(delete=False) as filter_file:
```
Comment (diox) [inline src/olympia/blocklist/cron.py]: Why keep it around ?

File: src/olympia/blocklist/cron.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import olympia.core.logger
+from olympia.lib.kinto import KintoServer
+
+from .mlbf import generate_mlbf, get_mlbf_key_format
+from .utils import KINTO_BUCKET, KINTO_COLLECTION_MLBF
+
+log = olympia.core.logger.getLogger('z.cron')
+
+
+def upload_mlbf_to_kinto():
```
Comment (diox) [inline src/olympia/blocklist/cron.py]: This should probably be behind a waffle switch for now.

File: src/olympia/blocklist/mlbf.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            for version, (version_id, _) in block.addon_versions.items()
+            if is_all_versions or block.is_version_blocked(version)}
+        all_versions.update(versions)
+    return all_versions.values()
+
+
+def get_all_guids():
+    from olympia.versions.models import Version
+
+    return Version.objects.values_list('addon__guid', 'version')
```
Comment (diox) [inline src/olympia/blocklist/mlbf.py]: You want deleted versions as well, I think. So `Version.unfiltered`.

Comment (eviljeff) [reply src/olympia/blocklist/mlbf.py]: https://github.com/mozilla/addons-server/pull/13659/commits/d34fd01b586cafd8a53855d9b5dc1c1e9056074a

File: src/olympia/blocklist/cron.py (hunk 2 above)
Comment (eviljeff) [reply src/olympia/blocklist/cron.py]: https://github.com/mozilla/addons-server/pull/13659/commits/d34fd01b586cafd8a53855d9b5dc1c1e9056074a

File: src/olympia/blocklist/cron.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/blocklist/cron.py]: https://github.com/mozilla/addons-server/pull/13659/commits/d34fd01b586cafd8a53855d9b5dc1c1e9056074a

File: src/olympia/blocklist/mlbf.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+
+def hash_filter_inputs(input_list, key_format):
+    return [
+        key_format.format(guid=guid, version=version)
+        for (guid, version) in input_list]
+
+
+def get_mlbf_key_format(salt=None):
+    salt = salt or secrets.randbits(128)
```
[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/blocklist/mlbf.py]: https://github.com/mozilla/addons-server/pull/13659/commits/d34fd01b586cafd8a53855d9b5dc1c1e9056074a

File: src/olympia/blocklist/management/commands/export_blocklist.py [hunk 5]
```diff
@@ -23,20 +20,15 @@ class Command(BaseCommand):
...
         """Handle command arguments."""
         parser.add_argument(
-            '--capacity',
-            type=float,
-            default='1.1',
-            dest='capacity',
-            help='MLBF capacity.')
+            '--salt',
+            type=int,
+            default=secrets.randbits(128),
```
[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/blocklist/management/commands/export_blocklist.py]: I think I'd rather keep it around as a way of debugging the filter generation, but I removed the default as suggested

**Comment by diox** [comment]
It's fine in this case, but note that I had not approved the PR yet

**Comment by eviljeff** [comment]
> It's fine in this case, but note that I had not approved the PR yet

d'oh!  Sorry, it looked like a full review.

**Pull request merged**